site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,115 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField from "@mui/material/TextField";
+import { useState, type FC } from "react";
+
+type TimeUnit = "days" | "hours";
+
+// Value should be in milliseconds or undefined. Undefined means no value.
+type DurationValue = number | undefined;
+
+type DurationFieldProps = {
+  label: string;
+  value: DurationValue;
+  onChange: (value: DurationValue) => void;
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const { label, value, onChange } = props;
+  const [timeUnit, setTimeUnit] = useState<TimeUnit>(() => {
+    if (!value) {
+      return "hours";
+    }
+
+    return Number.isInteger(durationToDays(value)) ? "days" : "hours";
+  });
```

Comment:
Just thought of another edge case. Because `DurationField` doesn't have its own `value` and always receives it from the parent, there's a risk that re-renders from a parent could cause state mismatches

1. We mount `DurationField` with a `value` of 48 hours in milliseconds
2. The value of `timeUnit` is set to `"days"` because `value` is cleanly divisible by 24 hours
3. Now let's say that somewhere above `DurationField`, `value` changes to be 16 hours in milliseconds
4. Now we have milliseconds that should be treated like hours, but `timeUnit` is still set to `"days"`

Reply:
Good catch! I'm going to handle this case.

Reply:
I thought using `useEffect` would solve the problem but it would cause a loop of course 😅. Since we are the only ones using this component I think we can live with this edge case for now. Wdyt? Do you see any options we could do without adding too much complexity? My concern is adding complexity to something that we will probably never face.

Reply:
I think the issue with `key` would be the same with `useEffect` right? When the `onChange` is called, the value is updated so the component would re-render re-calculating the suggested time unit even without the user doing it. Eg. I want to use 48 hours instead of 2 days, but in this case, when typing 48 it would set the dropdown to days and use 2 days instead. As a user, I would not expect things to get switched while typing.

Reply:
Oh, yeah – good call. Yeah, refreshing the state from scratch does have the risk of making the dropdown flip-flop between units without the user doing anything, which could be confusing/frustrating

I've got a couple of other ideas for how to avoid the state sync issue, but I think they'd actually be even more complicated. I'll wait until I do the test drive before deciding if they're actually needed